spec/models/manageiq/providers/amazon/aws_refresher_spec_counts.rb

```diff
@@ -0,0 +1,273 @@
+  end
+
+  def assert_ems(expected_table_counts)
+    expect(@ems).to have_attributes(
+      :api_version => nil, # TODO: Should be 3.0
+      :uid_ems     => nil
+    )
+    expect(@ems.flavors.size).to eql(expected_table_counts[:flavor])
+    expect(@ems.availability_zones.size).to eql(expected_table_counts[:availability_zone])
+    expect(@ems.vms_and_templates.size).to eql(expected_table_counts[:vm_or_template])
+    expect(@ems.security_groups.size).to eql(expected_table_counts[:security_group])
+    expect(@ems.network_ports.size).to eql(expected_table_counts[:network_port])
+    expect(@ems.cloud_networks.size).to eql(expected_table_counts[:cloud_network])
+    expect(@ems.floating_ips.size).to eql(expected_table_counts[:floating_ip])
+    expect(@ems.network_routers.size).to eql(expected_table_counts[:network_router])
+    expect(@ems.cloud_subnets.size).to eql(expected_table_counts[:cloud_subnet])
+    expect(@ems.miq_templates.size).to eq(expected_table_counts[:miq_template])
+
+    expect(@ems.orchestration_stacks.size).to eql(expected_table_counts[:orchestration_stack])
+  end
+
+  private
+
+  def manager
+    @ems
+  end
+
+  def aws_ec2
+    @aws_ec2 ||= manager.connect
+  end
+
+  def aws_cloud_formation
+    @aws_cloud_formation ||= manager.connect(:service => :CloudFormation)
+  end
+
+  def aws_elb
+    @aws_elb ||= manager.connect(:service => :ElasticLoadBalancing)
+  end
+
+  def aws_s3
+    @aws_s3 ||= manager.connect(:service => :S3)
+  end
+
+  def hash_collection
+    ::ManageIQ::Providers::Amazon::Inventory::HashCollection
+  end
+
+  # helpers for getting records fro mthe API
+  def instances
+    hash_collection.new(aws_ec2.instances)
+  end
+
+  def flavors
+    ManageIQ::Providers::Amazon::InstanceTypes.all
+  end
+
+  def availability_zones
+    hash_collection.new(aws_ec2.client.describe_availability_zones[:availability_zones])
+  end
+
+  def key_pairs
+    hash_collection.new(aws_ec2.client.describe_key_pairs[:key_pairs])
+  end
+
+  def private_images
+    hash_collection.new(aws_ec2.client.describe_images(:owners  => [:self],
+                                                       :filters => [{:name   => "image-type",
+                                                                     :values => ["machine"]}])[:images])
+  end
+
+  def shared_images
+    hash_collection.new(aws_ec2.client.describe_images(:executable_users => [:self],
+                                                       :filters          => [{:name   => "image-type",
+                                                                              :values => ["machine"]}])[:images])
+  end
+
+  def stacks
+    hash_collection.new(aws_cloud_formation.client.describe_stacks[:stacks])
+  end
+
+  def stacks_resources
+    resources = []
+    stacks.each do |stack|
+      resources << hash_collection.new(aws_cloud_formation.client.list_stack_resources(
+        :stack_name => stack["stack_name"]
+).try(:stack_resource_summaries)
```

Comment:
I bow to the robot overlords :-)

hopefully rubocop -a will fix it later, might be an issue of the rubocop-git